pkg/testgridanalysis/testreportconversion/testresult.go

```diff
@@ -10,3 +17,100 @@ func (a testResultsByPassPercentage) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
 func (a testResultsByPassPercentage) Less(i, j int) bool {
 	return a[i].PassPercentage < a[j].PassPercentage
 }
+
+func combineTestResults(lhs, rhs []sippyprocessingv1.TestResult) []sippyprocessingv1.TestResult {
+	byTestName := map[string]sippyprocessingv1.TestResult{}
+
+	for i := range lhs {
+		currTestResult := lhs[i]
+		byTestName[currTestResult.Name] = currTestResult
+	}
+
+	for i := range rhs {
+		currTestResult := rhs[i]
+		existing, ok := byTestName[currTestResult.Name]
+		if !ok {
+			byTestName[currTestResult.Name] = currTestResult
+			continue
+		}
+
+		existing.Failures += currTestResult.Failures
+		existing.Successes += currTestResult.Successes
+		existing.Flakes += currTestResult.Flakes
+		existing.PassPercentage = percent(existing.Successes, existing.Failures)
+		// bugs should be the same for now.
+		byTestName[currTestResult.Name] = existing
+	}
+
+	combined := []sippyprocessingv1.TestResult{}
+	for _, currTestResult := range byTestName {
+		combined = append(combined, currTestResult)
+	}
+
+	sort.Stable(testResultsByPassPercentage(combined))
```

Comment:
given the way you're using this (invoking combine for each jobresult) i would have separated the sort so you aren't repeatedly sorting the intermediary results you don't care about as you build up the aggregated list.

Reply:
> given the way you're using this (invoking combine for each jobresult) i would have separated the sort so you aren't repeatedly sorting the intermediary results you don't care about as you build up the aggregated list.

Depends on where and how it's called.  I'd like to keep the data clean when combining two (usually, might even be always now) sorted lists.  if we find ourselves getting two slow, I'm confident we could deal with it then.